tests/functional/aws-node-sdk/test/versioning/multiObjectDelete.js

```diff
@@ -138,3 +139,96 @@ testing('Multi-Object Versioning Delete Success', function success() {
         });
     });
 });
+
+testing('Multi-Object Versioning Delete - deleting delete marker',
+() => {
+    withV4(sigCfg => {
+        const bucketUtil = new BucketUtility('default', sigCfg);
+        const s3 = bucketUtil.s3;
+
+        beforeEach(done => {
+            async.waterfall([
+                next => s3.createBucket({ Bucket: bucketName },
+                    err => next(err)),
+                next => s3.putBucketVersioningAsync({
+                    Bucket: bucketName,
+                    VersioningConfiguration: {
+                        Status: 'Enabled',
+                    },
+                }, err => next(err)),
+            ], done);
+        });
+        afterEach(done => {
+            removeAllVersions({ Bucket: bucketName }, err => {
+                if (err) {
+                    return done(err);
+                }
+                return s3.deleteBucket({ Bucket: bucketName }, err => {
+                    assert.strictEqual(err, null,
+                        `Error deleting bucket: ${err}`);
+                    return done();
+                });
+            });
+        });
+
+        it('deleteing a delete marker - should send back VersionId and ' +
+        'DeleteMarkerVersionId equals to deleteVersionId', done => {
+            async.waterfall([
+                next => s3.putObject({ Bucket: bucketName, Key: key },
+                  err => next(err)),
+                next => s3.deleteObject({ Bucket: bucketName,
+                Key: key }, (err, data) => {
+                    const deleteVersionId = data.VersionId;
+                    next(err, deleteVersionId);
+                }),
+                (deleteVersionId, next) => s3.deleteObjects({ Bucket:
+                  bucketName,
+                  Delete: {
+                      Objects: [
+                          {
+                              Key: key,
+                              VersionId: deleteVersionId,
+                          },
+                      ],
+                  } }, (err, data) => {
+                    assert(data.Deleted[0].DeleteMarker);
+                    assert.strictEqual(data.Deleted[0].VersionId,
+                      deleteVersionId);
+                    assert.strictEqual(data.Deleted[0].DeleteMarkerVersionId,
+                      deleteVersionId);
+                    next(err);
+                }),
+            ], err => done(err));
+        });
+
+        it('delete object that has not been created - should send back the ' +
```

Comment:
should send back a DeleteMarkerVersionId matching the versionId stored for the object